### ISSUE TITLE
[Bug]: B-Frames are not generated on RX 7900 XTX Ubuntu 22.04

### ISSUE THREAD

**Issue by RFCArrow** [GPUOpen-LibrariesAndSDKs/AMF#403]
**Describe the bug**
B-frames are not generated by AMF. In the simplest case, the SimpleEncoder sample project does not produce any b-frames.

**To Reproduce**
Apply the following patch to prevent b-frames from being disabled:
```
diff --git a/amf/public/samples/CPPSamples/SimpleEncoder/SimpleEncoder.cpp b/amf/public/samples/CPPSamples/SimpleEncoder/SimpleEncoder.cpp
index d4ed237..411c6be 100644
--- a/amf/public/samples/CPPSamples/SimpleEncoder/SimpleEncoder.cpp
+++ b/amf/public/samples/CPPSamples/SimpleEncoder/SimpleEncoder.cpp
@@ -52,7 +52,7 @@
 
 static const bool bEnable4K = false;
 static const bool bEnable10bit = false;
-static const bool bMaximumSpeed = true; // encoding speed preset
+static const bool bMaximumSpeed = false; // encoding speed preset
 
 static const bool bEnableEfc = false; // color conversion inside encoder component. Will use GFX or HW if available
 // Encoder Format Conversion (EFC) available for selected input formats
```

Then build and run the SimpleEncoder project:
`cd AMF/amf`
`VK_SDK_PATH=~/vulkan/1.3.239.0 make -C public/samples`
`./bin/dbg_64/SimpleEncoder`

Inspecting the output reveals no B frames are generated
`ffprobe -i output.h264 -show_frames | grep pict_type`

```
pict_type=I
pict_type=P
pict_type=P
pict_type=P
pict_type=P
pict_type=P
[...]
```

**Setup (please complete the following information):**
OS: Ubuntu Linux 22.04 LTS (5.19.0-45-generic)
GPU: RX 7900 XTX
Driver Version:
```
amdgpu         1:5.4.50406-1580598.22.04 amd64        Meta package to install amdgpu components.
libamdenc-amdgpu-pro:amd64 1.0-1580631.22.04 amd64        AMD Encode Core Library
amf-amdgpu-pro:amd64 1.4.29-1580631.22.04 amd64        AMDGPU Pro Advanced Multimedia Framework
```
AMF Source: 1.4.29 origin/master (https://github.com/GPUOpen-LibrariesAndSDKs/AMF/commit/bd5db31d3d8ea1fae7002dd903898599bdb7d97a)
Which component has the issue: Encoder

**Debug Log (please upload or paste):**
```
[AMFDecoderUVDImpl]   Error: ..\..\..\..\..\..\runtime\src\components\DecoderUVD\DecoderUVDImpl.cpp(1996):AMF_ERROR 1 : AMF_FAIL: pCaps->Init(m_pContext, m_codecId)
```

**Expected behavior**
B-frames exist in the output bitstream.

**Additional context**
I also tried using the sample `TranscodeHW` project (no changes applied) on a a different content in case the the SimpleEncoder generated content is not complex enough to get the encoder to produce b-frames.

I used the following command line
`./bin/dbg_64/TranscodeHW -INPUT ~/assets/other/input_1920x1080_8bit_420p.yuv -OUTPUT transcode_output.h264 -Usage TRANSCODING -MaxConsecutiveBPictures 3 -TRACE_LEVEL 3`
 I have attached the generated debug log [here](https://github.com/GPUOpen-LibrariesAndSDKs/AMF/files/12163310/debug_log.txt)

**Comment by RFCArrow** [comment]
It may also be worth noting that the `CapabilityManager` sample project shows that B pictures are supported for the AVC encoder on this device.
```
Querying video encoder capabilities...
    Codec AMFVideoEncoderVCE_AVC
        Number HW instances:2
        --------------------
        Instance:0
        Acceleration Type:Hardware-accelerated
        maximum profile:100
        maximum level:52
        Number of temporal Layers:4
        IsBPictureSupported:1
        Max Number of streams supported:16
        Encoder input:
            Width: [128-4096]
            Height: [128-4096]
            Vertical alignment: 32 lines.
            Interlaced support: NO
            Total of 6 pixel format(s) supported:
                0: YUV420P 
                1: YV12 
                2: BGRA 
                3: RGBA 
                4: ARGB 
                5: NV12 (native)
            Total of 3 memory type(s) supported:
                0: OPENCL 
                1: OPENGL 
                2: HOST 
        Encoder output:
            Width: [128-4096]
            Height: [128-4096]
            Vertical alignment: 32 lines.
            Interlaced support: NO
            Total of 1 pixel format(s) supported:
                0: NV12 (native)
            Total of 4 memory type(s) supported:
                0: XV (native)
                1: OPENCL 
                2: OPENGL 
                3: HOST 
        --------------------
        Instance:1
        Acceleration Type:Hardware-accelerated
        maximum profile:100
        maximum level:52
        Number of temporal Layers:4
        IsBPictureSupported:1
        Max Number of streams supported:16
        Encoder input:
            Width: [128-4096]
            Height: [128-4096]
            Vertical alignment: 32 lines.
            Interlaced support: NO
            Total of 6 pixel format(s) supported:
                0: YUV420P 
                1: YV12 
                2: BGRA 
                3: RGBA 
                4: ARGB 
                5: NV12 (native)
            Total of 3 memory type(s) supported:
                0: OPENCL 
                1: OPENGL 
                2: HOST 
        Encoder output:
            Width: [128-4096]
            Height: [128-4096]
            Vertical alignment: 32 lines.
            Interlaced support: NO
            Total of 1 pixel format(s) supported:
                0: NV12 (native)
            Total of 4 memory type(s) supported:
                0: XV (native)
                1: OPENCL 
                2: OPENGL 
                3: HOST 
 ```

**Comment by rhutsAMD** [comment]
To enable B frames, both `AMF_VIDEO_ENCODER_MAX_CONSECUTIVE_BPICTURES` and `AMF_VIDEO_ENCODER_B_PIC_PATTERN` should be set. Please try setting `-BPicturesPattern <value>` as well inside your TranscodeHW command line arguments.

**Comment by rhutsAMD** [comment]
Thank you for noticing this. The documentation has been corrected in commit 2f326350e849894a929296854f5290e66197c97c.
